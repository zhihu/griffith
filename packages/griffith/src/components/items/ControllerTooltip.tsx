import React from 'react'
import {css} from 'aphrodite/no-important'
import {LocaleConfigKey} from '../../constants/locales'
import useBoolean from '../../hooks/useBoolean'
import TranslatedText from '../TranslatedText'
import styles from '../Controller.styles'

type Props = {
  content: LocaleConfigKey
}

const canUseTouch =
  typeof document !== 'undefined' && 'ontouchstart' in document.body

const ControllerTooltip: React.FC<Props> = ({content, children}) => {
  const [isHovered, isHoveredSwitch] = useBoolean()

  return (
    <div
      className={css(styles.menuContainer)}
      onMouseEnter={isHoveredSwitch.on}
      onMouseLeave={isHoveredSwitch.off}
    >
      {children}
      {!canUseTouch && (
        <div
          className={css(
            styles.menu,
            isHovered && styles.menuShown,
            styles.tooltipContent
          )}
        >
          <TranslatedText name={content} />
        </div>
      )}
    </div>
  )
}

export default ControllerTooltip
