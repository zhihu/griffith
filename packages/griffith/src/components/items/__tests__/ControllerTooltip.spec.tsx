import React from 'react'
import Renderer from 'react-test-renderer'
import ControllerTooltip from '../ControllerTooltip'

describe('Tooltip', () => {
  it('get Tooltip component', () => {
    expect(
      Renderer.create(<ControllerTooltip content="quality-hd" />).toJSON()
    ).toMatchSnapshot()
  })
})
