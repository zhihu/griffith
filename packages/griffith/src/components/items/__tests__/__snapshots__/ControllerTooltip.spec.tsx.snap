// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tooltip get Tooltip component 1`] = `
<div
  className="menuContainer_e296pg"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <div
    className="menu_1a7s4a3-o_O-tooltipContent_me14v4"
  >
    HD
  </div>
</div>
`;
